clientconn_test.go

```diff
@@ -820,3 +820,128 @@ func TestBackoffCancel(t *testing.T) {
 	cc.Close()
 	// Should not leak. May need -count 5000 to exercise.
 }
+
+type signalingBackoffStrategy struct {
+	ready chan struct{}
+}
+
+func (bs *signalingBackoffStrategy) Backoff(retries int) time.Duration {
```

Comment:
This doesn't seem like the right way to do this..  We consult the `Backoff()` at the start of an iteration, and it's just a coincidence that we're setting the state to `Connecting` _after_ calling `Backoff()`.  If those were flipped, then you would probably miss the `TransientFailure` state, or at least it would be a race.

I was picturing a backoff time of <minutes>, and the signal to unblock would be a call to `cc.ResetConnectBackoff()`

Reply:
Done.